vertx-core/src/main/java/io/vertx/core/net/impl/ConnectionBase.java

```diff
@@ -390,7 +392,7 @@ public void flushBytesWritten() {
   }
 
   public boolean isSsl() {
-    return chctx.pipeline().get(SslHandler.class) != null;
+    return chctx.pipeline().get(SslHandler.class) != null || (chctx.channel().parent() != null && chctx.channel().parent().pipeline().get(SslHandlerWrapper.class) != null);
```

Comment:
instead we should use inheritance for that and override in `Http3ConnectionImpl`